docs/docs/data-structures/tree/transactions.mdx

```diff
@@ -0,0 +1,34 @@
+---
+title: Transactions
+sidebar_position: 6
+---
+
+If you want the `SharedTree` to treat a set of changes atomically, wrap these changes in a transaction.
+Using a transaction guarantees that (if applied) all of the changes will be applied together synchronously (though, note that the Fluid Framework guarantees this already for any sequence of changes that are submitted synchronously). However, the changes may not be applied at all if the transaction is given one or more constraints.
+If any constraint on a transaction is not met, then the transaction and all its changes will ignored by all clients.
```

Comment:
It also needs to include a section about nested transactions similar to how it is described [here](https://github.com/microsoft/FluidFramework/blob/9bdb9b5d606d9312f3679cf501edff5c8dd68537/packages/dds/tree/src/simple-tree/api/tree.ts#L574).